cmd/prompt.go

```diff
@@ -0,0 +1,402 @@
+	switch annotation {
+	case "XPATH":
+		word := doc.GetWordBeforeCursor()
+		suggestions := make([]goprompt.Suggest, 0, 16)
+		for _, entry := range schemaTree.Dir {
+			suggestions = append(suggestions, findMatchedXPATH(entry, word, 0)...)
+		}
+		return suggestions
+	case "FILE":
+		return filePathCompleter.Complete(doc)
+	case "YANG":
+		return yangPathCompleter.Complete(doc)
+	case "MODEL":
+		suggestions := make([]goprompt.Suggest, 0, len(schemaTree.Dir))
+		for name := range schemaTree.Dir {
+			suggestions = append(suggestions, goprompt.Suggest{Text: name})
+		}
+		return goprompt.FilterHasPrefix(suggestions, doc.GetWordBeforeCursor(), true)
+	case "DIR":
+		return dirPathCompleter.Complete(doc)
+	}
+	return []goprompt.Suggest{}
+}
+
+func showCommandArguments(b *goprompt.Buffer) {
+	doc := b.Document()
+	showLocalFlags := false
+	command := rootCmd
+	args := strings.Fields(doc.CurrentLine())
+	if found, _, err := command.Find(args); err == nil {
+		if command != found {
+			showLocalFlags = true
+		}
+		command = found
+	}
+	maxNameLen := 0
+	suggestions := make([]goprompt.Suggest, 0, 32)
+	if command.HasAvailableSubCommands() {
+		for _, c := range command.Commands() {
+			if c.Hidden {
+				continue
+			}
+			length := len(c.Name())
+			if maxNameLen < length {
+				maxNameLen = length
+			}
+			suggestions = append(suggestions, goprompt.Suggest{Text: c.Name(), Description: c.Short})
+		}
+	}
+	if showLocalFlags {
+		addFlags := func(flag *pflag.Flag) {
+			if flag.Hidden {
+				return
+			}
+			length := len(flag.Name)
+			if maxNameLen < length+2 {
+				maxNameLen = length + 2
+			}
+			suggestions = append(suggestions, goprompt.Suggest{Text: "--" + flag.Name, Description: flag.Usage})
+		}
+		command.LocalFlags().VisitAll(addFlags)
+	}
+	suggestions = goprompt.FilterHasPrefix(suggestions, doc.GetWordBeforeCursor(), true)
+	if len(suggestions) == 0 {
+		return
+	}
+	if err := termbox.Init(); err != nil {
+		logger.Fatalf("%v", err)
+	}
+	w, _ := termbox.Size()
+	termbox.Close()
+	fmt.Printf("\n")
+	maxDescLen := w - maxNameLen - 6
+	format := fmt.Sprintf("  %%-%ds : %%-%ds\n", maxNameLen, maxDescLen)
+	for i := range suggestions {
+		length := len(suggestions[i].Description)
+		if length > maxDescLen {
+			fmt.Printf(format, suggestions[i].Text, suggestions[i].Description[:maxDescLen])
+		} else {
+			fmt.Printf(format, suggestions[i].Text, suggestions[i].Description)
+		}
+	}
+	fmt.Printf("\n")
+}
+
+// ExecutePrompt load and run gnmic-prompt mode.
+func ExecutePrompt() {
+	var err error
+	schemaTree, err = loadSchemaZip()
+	if err != nil {
+		schemaTree = buildRootEntry()
+	}
+	rootCmd.AddCommand(promptQuitCmd)
+	rootCmd.RemoveCommand(promptModeCmd)
+	shell := &cmdPrompt{
+		RootCmd: rootCmd,
+		GoPromptOptions: []goprompt.Option{
+			goprompt.OptionTitle("gnmic-prompt"),
+			goprompt.OptionPrefix("gnmic> "),
+			goprompt.OptionHistory(promptHistory),
+			goprompt.OptionMaxSuggestion(5),
```

Comment:
can you increase MaxSuggestions to at least 15, to fit all possible flags for commands like set and subscribe?

Reply:
That would be get a message, "Your console window is too small...", from go-prompt package.
MaxSuggestion will make a limitation for the console size.

Reply:
OK, then let's keep it and we can think about setting a dynamic suggestions length based on the terminal size later on.